lib/school_of_fish.rb

```diff
@@ -24,22 +34,88 @@ def update(dt)
     elsif (rand<0.01)
       @y_fish_speed = 1*rand-0.5
     end
+
+    check_distance_to(@player)
+
+    if(@dispersing)
+      decrease_density(dt)
```

Comment:
I recommend moving the disperse behavior to its own method to keep the `update` method clean